### PR TITLE
#716 Implement GitlabStars

### LINKED ISSUE #716: Implement GitlabStars
Similarly to how Github repo Stars are implemented, we should also implemented them for GitLab.

### PR DESCRIPTION
**Pull request by fellahi-ali**
Fixes #716

### CODE REVIEW

**Comment by charlesmike** [comment]
@fellahi-ali thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2020-12-10T15:35:49.655735``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: self-core-impl/src/test/java/com/selfxdsd/core/GitlabRepoTestCase.java [hunk 1]
```diff
@@ -159,5 +160,39 @@ public void returnsLabels() {
...
+            Matchers.allOf(
+                Matchers.notNullValue(),
+                Matchers.instanceOf(GitlabStars.class)
+            )
+        );
+        repo.stars().add();
+        MatcherAssert.assertThat(
+            resources.requests().last().getUri().toString(),
+            Matchers.equalTo("https://gitlab.com/projects/test/repo/star")
+        );
```
Comment (amihaiemil) [inline self-core-impl/src/test/java/com/selfxdsd/core/GitlabRepoTestCase.java:196]: @fellahi-ali You can remove this assert and ``repo.starts().add()``. In this test, we are only interested to see if GitlabRepo returns the proper instance of Stars.

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@fellahi-ali Just one comment. 

And please check Travis, Checkstyle has a few complaints :D

File: self-core-impl/src/test/java/com/selfxdsd/core/GitlabRepoTestCase.java (hunk 1 above)
Comment (fellahi-ali) [reply self-core-impl/src/test/java/com/selfxdsd/core/GitlabRepoTestCase.java:196]: @amihaiemil Sure! actually I felt I'm overthinking there :(

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**